### PR TITLE
feat: warn on unknown output type

### PR DESCRIPTION
**Pull request by Weakky**

### CODE REVIEW

File: src/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export type OnUnknownFieldName = (info: {
+  unknownFieldName: string
+  error: Error
+  validFieldNames: string[]
+  typeName: string
+}) => void
+
+export type OnUnknownOutputType = (info: {
```
Comment (jasonkuhrt) [inline src/hooks.ts]:
```suggestion
export type OnUnknownFieldType = (info: {
```

Might make more sense

File: src/graphql.ts [hunk 2]
```diff
@@ -27,3 +27,5 @@ export const scalarsNameValues: ScalarName[] = [
   'ID',
   'Boolean',
 ]
+
+export const isScalarType = (name: string) => scalarsNameValues.includes(name as any)
```
Comment (jasonkuhrt) [inline src/graphql.ts]:
```suggestion
export const isScalarType = (name: string): boolean => scalarsNameValues.includes(name as any)
```

File: src/hooks.ts (hunk 1 above)
Comment (jasonkuhrt) [reply src/hooks.ts]: Other instances of this in the PR would need changing too then.

File: src/builder.ts [hunk 3]
```diff
@@ -335,6 +346,13 @@ export class SchemaBuilder {
           }
           const fieldName = resolvedConfig.alias || field.name
           const type = resolvedConfig.type || field.outputType.type
+
+          if (
+            !this.assertOutputTypeIsDefined(typeName, fieldName, type, stage)
```
Comment (jasonkuhrt) [inline src/builder.ts]: Can we use naming to clarify the difference between `typeName` and `type`. e.g. `modelType`? What is it?

Comment (jasonkuhrt) [inline src/builder.ts]:
```suggestion
            !this.assertOutputTypeIsDefined(fieldType, fieldName, type, stage)
```

And instances later. Think its clearer to associate that this type name is the type name of the field type.

**Review by jasonkuhrt — APPROVED** [review]

File: src/builder.ts (hunk 3 above)
Comment (Weakky) [reply src/builder.ts]: yeah we can! it makes a bit more sense when you see where `type` is going (in the `type` property of `t.field`). Here `type` is the OutputType, and `typeName`, the parent typename in which the dynamic is called. I'll find better names!

**Pull request merged**